src/mmu/timers.rs

```diff
@@ -0,0 +1,292 @@
+
+        // The system clock is a 16-bit number in t-cycles, but it is only incremented once every m-cycle.
+        if self.timers.system_clock_counter == T_CYCLES_PER_M_CYCLE {
+            self.timers.system_clock_counter = 0;
+            self.timers.system_clock = self.timers.system_clock.wrapping_add(T_CYCLES_PER_M_CYCLE);
+        }
+
+        // ----- Updating DIV
+        let div_value = (self.timers.system_clock >> 8) as u8;
+        self.write_byte_override(DIV_ADDR, div_value);
+
+        // ----- Updating TIMA
+        let tac_enable = self.get_tac_enable();
+        let tima_bit_is_active = self.get_tima_bit_state(false);
+        let tima_bit_was_active = self.get_tima_bit_state(true);
+
+        // TIMA's special overflow behavior occurs the cycle AFTER an overflow is detected.
+        if self.timers.tima_overflowed {
+            self.process_tima_overflow();
+        }
+
+        // TIMA increments while enabled, and on a falling clock edge. The second condition is
+        // there because memory writes can trigger a clock edge detection that occurs elsewhere.
+        if tac_enable && tima_bit_was_active && !tima_bit_is_active
+            || self.timers.tima_falling_edge_detected
+        {
+            self.increment_tima();
+            self.timers.tima_falling_edge_detected = false;
+        }
+
+        // It's important to update the previous clockstate here, instead of at the beginning of the loop.
+        // This is because the system clock can be reset if something writes to the div timer.
+        // Therefore, the system clock state might be different by the next time this function is called.
+        self.timers.system_clock_prev = self.timers.system_clock;
+    }
+
+    fn increment_tima(&mut self) {
+        let mut tima_byte = self.read_byte(TIMA_ADDR);
+        let overflowed;
+
+        (tima_byte, overflowed) = tima_byte.overflowing_add(1);
+
+        self.write_byte_override(TIMA_ADDR, tima_byte);
+
+        if overflowed {
+            self.timers.tima_overflowed = true;
+        }
+    }
+
+    /// Overflows of TIMA send a timer interrupt. When TIMA overflows, its value is set
+    /// to the value stored in TMA.
+    fn process_tima_overflow(&mut self) {
+        self.timers.tima_overflowed = false;
+        self.timers.tima_write_lock_counter = TIMA_WRITE_LOCK_T_CYCLES;
+
+        self.request_interrupt(TIMER_INTERRUPT_BIT);
+        let tma_value = self.read_byte(TMA_ADDR);
+        self.write_byte_override(TIMA_ADDR, tma_value);
+    }
+
+    fn get_tac_enable(&self) -> bool {
+        let byte = self.read_byte(TAC_ADDR);
+        get_bit(byte, TAC_ENABLE_BIT)
+    }
+
+    /// The clock select bits of the TAC register control which bit of the global timer
+    /// that TIMA tracks. This effectively selects the frequency at which TIMA will increment.
+    fn get_system_clock_bit_for_tima(&self) -> u8 {
+        let byte = self.read_byte(TAC_ADDR);
+        let value = byte & 0b_0000_0011;
```

Comment:
You did nothing wrong here, and it's very readable.  If you are looking at other code, keep in mind , it's considered perfectly legible to use 0x03 as a bitmask.  Hex is known as "shorthand binary" because if you memorize the bit patterns from 0-15 you can easily decode back and forth from binary.  Not that I have, but by the end of this project you might!

 If you need it in another position in the byte, say shifted 2 over,  you might use `0x03 << 2` which might be a bit more legible than 0x0C.